### PR TITLE
catch additional zk connection errors while creating table

### PR DESCRIPTION
**Pull request by sum12**
and handle the case of dropping empty table dirs

incase of erros in those two lines the dirs are left behind and retries start failling 

### Changelog category (leave one):
- Not for changelog (changelog entry is not required)

### Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
catch additional zk connection erros while creating table and make sure to cleanup dirs if necessary for retries

### Documentation entry for user-facing changes

- [ ] Documentation is written (mandatory for new features)

> Information about CI checks: https://clickhouse.com/docs/en/development/continuous-integration/

#### CI Settings (Only check the boxes if you know what you are doing):
- [ ]  Allow: All Required Checks
- [ ]  Allow: Stateless tests
- [ ]  Allow: Stateful tests
- [ ]  Allow: Integration Tests
- [ ]  Allow: Performance tests
- [ ]  Allow: All Builds
- [ ]  Allow: batch 1, 2 for multi-batch jobs
- [ ]  Allow: batch 3, 4, 5, 6 for multi-batch jobs
---
- [ ]  Exclude: Style check
- [ ]  Exclude: Fast test
- [ ]  Exclude: All with ASAN
- [ ]  Exclude: All with TSAN, MSAN, UBSAN, Coverage
- [ ]  Exclude: All with aarch64, release, debug
---
- [ ]  Run only fuzzers related jobs (libFuzzer fuzzers, AST fuzzers, etc.)
- [ ]  Exclude: AST fuzzers
---
- [ ]  Do not test
- [ ]  Woolen Wolfdog
- [ ]  Upload binaries for special builds
- [ ]  Disable merge-commit
- [ ]  Disable CI cache

### CODE REVIEW

File: src/Storages/StorageReplicatedMergeTree.cpp [hunk 1]
```diff
@@ -552,9 +556,6 @@ StorageReplicatedMergeTree::StorageReplicatedMergeTree(
         throw;
     }
 
-    createNewZooKeeperNodes();
-    syncPinnedPartUUIDs();
-
     if (!has_metadata_in_zookeeper.has_value() || *has_metadata_in_zookeeper)
         createTableSharedID();
```
Comment (tavplubix) [inline src/Storages/StorageReplicatedMergeTree.cpp:560]: should we move this as well?

**Comment by robot-clickhouse-ci-2** [comment]
*This is an automated comment for commit f0490e83729be2ddafa52455f0a52a3e32619ffb with description of existing statuses. It's updated for the latest CI running*

[✅ Click here](https://s3.amazonaws.com/clickhouse-test-reports/69093/f0490e83729be2ddafa52455f0a52a3e32619ffb/ci_running.html) to open a full report in a separate page

<details><summary>Successful checks</summary>
<table>
<thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody>
<tr><td>Docs check</td><td>Builds and tests the documentation</td><td>✅ success</td></tr>
<tbody>
</table>
</details>

File: src/Storages/StorageReplicatedMergeTree.cpp (hunk 1 above)
Comment (sum12) [reply src/Storages/StorageReplicatedMergeTree.cpp]: sure, moved them

**Comment by sum12** [comment]
Can we move this ahead please.

**Comment by tavplubix** [comment]
AST fuzzer (asan)  - https://github.com/ClickHouse/ClickHouse/issues/57728

**Pull request merged**